provider/testdata/Cargo.toml

```diff
@@ -303,19 +303,7 @@ displaydoc = { version = "0.2.3", default-features = false, optional = true }
 writeable = { version = "0.4", path = "../../utils/writeable", optional = true }
 
 # databake deps
-icu_calendar = { version = "1.0.0-beta1", path = "../../components/calendar", optional = true }
-icu_casemapping = { version = "0.2", path = "../../experimental/casemapping", optional = true }
-icu_collator = { version = "1.0.0-beta1", path = "../../components/collator", optional = true }
-icu_datetime = { version = "1.0.0-beta1", path = "../../components/datetime", features = ["experimental"], optional = true }
-icu_decimal = { version = "1.0.0-beta1", path = "../../components/decimal", optional = true }
-icu_list = { version = "1.0.0-beta1", path = "../../components/list", optional = true }
-icu_locid_transform = { version = "1.0.0-beta1", path = "../../components/locid_transform", optional = true }
-icu_normalizer = { version = "1.0.0-beta1", path = "../../components/normalizer", optional = true }
-icu_plurals = { version = "1.0.0-beta1", path = "../../components/plurals", optional = true }
-icu_properties = { version = "1.0.0-beta1", path = "../../components/properties", optional = true }
-icu_segmenter = { version = "1.0.0-alpha1", path = "../../experimental/segmenter", optional = true, features = ["lstm"] }
-icu_timezone = { version = "1.0.0-beta1", path = "../../components/timezone", optional = true }
-icu_collections = { version = "1.0.0-beta1", path = "../../components/collections", optional = true }
+icu = { version = "1.0.0-beta1", path = "../../components/icu", optional = true, features = ["experimental"] }
```

Comment:
Suggestion: I'm in favor of the flag on `datagen`, but I may prefer to keep it turned off for `icu_testdata`.

I'm not necessarily a fan of going through the metacrate more than we need to. We have it for client convenience, and we are using it in docs strings despite mild objections from myself. Using it here in testdata gains us nothing, and it doesn't get us closer to the goal of #2383, which is to make each component a separate feature on icu_testdata.

Besides, it's probably going to be easier for BakedDataExporter and/or DataBake to generate `#[cfg(feature = "icu_datetime")]`, since it knows the crate name. You don't want that to get caught up in your global search-and-replace.